core/src/main/scala/org/apache/spark/internal/config/ConfigBuilder.scala

```diff
@@ -199,6 +200,11 @@ private[spark] case class ConfigBuilder(key: String) {
     this
   }
 
+  def version(v: String): ConfigBuilder = {
```

Comment:
How could we know `v` is a valid version name? Why not make it an Enum?

Reply:
Although we can checked it, but I think convention is greater than configuration.
This is very similar to `ExpressionDescription.since`.
If we create an Enum, it has big size. So I don’t think it ’s appropriate to use Enum.